app/src/main/java/com/github/multimatum_team/multimatum/adaptater/UserGroupAdapter.kt

```diff
@@ -0,0 +1,102 @@
+package com.github.multimatum_team.multimatum.adaptater
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Typeface
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import com.github.multimatum_team.multimatum.R
+import com.github.multimatum_team.multimatum.model.GroupID
+import com.github.multimatum_team.multimatum.model.UserGroup
+import com.github.multimatum_team.multimatum.viewmodel.GroupViewModel
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+
+class UserGroupAdapter(context: Context,
+    private val groupViewModel: GroupViewModel
+): BaseAdapter() {
+    private var dataSource: List<UserGroup> = listOf()
+    private val inflater: LayoutInflater =
+        context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+
+
+    fun setGroups(groups: Map<GroupID, UserGroup>){
+        dataSource = groups.values.toList()
```

Comment:
It seems that you need to notify the adapter that the data has changed using `notifyDataSetChanged()` just after updating `dataSource`.